libfreerdp/core/input.c

```diff
@@ -218,37 +242,45 @@ void input_send_fastpath_unicode_keyboard_event(rdpInput* input, UINT16 flags, U
 
 	eventFlags |= (flags & KBD_FLAGS_RELEASE) ? FASTPATH_INPUT_KBDFLAGS_RELEASE : 0;
 	s = fastpath_input_pdu_init(rdp->fastpath, eventFlags, FASTPATH_INPUT_EVENT_UNICODE);
+	if (!s)
+		return FALSE;
 	Stream_Write_UINT16(s, code); /* unicodeCode (2 bytes) */
-	fastpath_send_input_pdu(rdp->fastpath, s);
+	return fastpath_send_input_pdu(rdp->fastpath, s);
 }
 
-void input_send_fastpath_mouse_event(rdpInput* input, UINT16 flags, UINT16 x, UINT16 y)
+BOOL input_send_fastpath_mouse_event(rdpInput* input, UINT16 flags, UINT16 x, UINT16 y)
 {
 	wStream* s;
 	rdpRdp* rdp = input->context->rdp;
 
 	s = fastpath_input_pdu_init(rdp->fastpath, 0, FASTPATH_INPUT_EVENT_MOUSE);
+	if (!s)
+		return FALSE;
 	input_write_mouse_event(s, flags, x, y);
-	fastpath_send_input_pdu(rdp->fastpath, s);
+	return fastpath_send_input_pdu(rdp->fastpath, s);
 }
 
-void input_send_fastpath_extended_mouse_event(rdpInput* input, UINT16 flags, UINT16 x, UINT16 y)
+BOOL input_send_fastpath_extended_mouse_event(rdpInput* input, UINT16 flags, UINT16 x, UINT16 y)
 {
 	wStream* s;
 	rdpRdp* rdp = input->context->rdp;
 
 	s = fastpath_input_pdu_init(rdp->fastpath, 0, FASTPATH_INPUT_EVENT_MOUSEX);
+	if (!s)
+		return FALSE;
 	input_write_extended_mouse_event(s, flags, x, y);
-	fastpath_send_input_pdu(rdp->fastpath, s);
+	return fastpath_send_input_pdu(rdp->fastpath, s);
 }
 
-void input_send_fastpath_focus_in_event(rdpInput* input, UINT16 toggleStates)
+BOOL input_send_fastpath_focus_in_event(rdpInput* input, UINT16 toggleStates)
 {
 	wStream* s;
 	rdpRdp* rdp = input->context->rdp;
 	BYTE eventFlags = 0;
 
 	s = fastpath_input_pdu_init_header(rdp->fastpath);
+	if (!s)
```

Comment:
fastpath_input_pdu_init_header returns a stream with 256 byte or FALSE. We only write 5 bytes plus headers here.